controllers/vpcendpoint/cleanup.go

```diff
@@ -35,36 +35,46 @@ func (r *VpcEndpointReconciler) cleanupAwsResources(ctx context.Context, resourc
 			return err
 		}
 
-		resp, err := r.awsClient.GetHostedZone(ctx, resource.Status.HostedZoneId)
-		if err != nil {
+		// Ensure .status.hostedZoneId is populated
+		if err := r.validateR53PrivateHostedZone(ctx, resource); err != nil {
 			return err
 		}
 
-		if resp.HostedZone != nil {
-			r.log.V(0).Info("Deleting Route53 Hosted Zone Record")
-			input := &route53Types.ResourceRecordSet{
-				Name:            aws.String(fmt.Sprintf("%s.%s", resource.Spec.CustomDns.Route53PrivateHostedZone.Record.Hostname, *resp.HostedZone.Name)),
-				ResourceRecords: []route53Types.ResourceRecord{*resourceRecord},
-				TTL:             aws.Int64(300),
-				Type:            route53Types.RRTypeCname,
+		// HostedZoneId and resourceRecord are required if we want to clean up a ResourceRecordSet
+		if resource.Status.HostedZoneId != "" && resourceRecord != nil {
```

Comment:
GitHub's showing this diff a little confusingly, so just want to clarify for ease of review I surrounded this block with this if statement and moved the earlier `GetHostedZone()` from line 38 inside this block as `GetHostedZone()` requires a non-empty id to be supplied.